src/librbd/image/CloneRequest.h

```diff
@@ -0,0 +1,152 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_IMAGE_CLONE_REQUEST_H
+#define CEPH_LIBRBD_IMAGE_CLONE_REQUEST_H
+
+#include "common/WorkQueue.h"
+#include "include/rados/librados.hpp"
+#include "include/rbd_types.h"
+#include "cls/rbd/cls_rbd_types.h"
+#include "include/rbd/librbd.hpp"
+#include "librbd/ImageCtx.h"
+
+class Context;
+
+using librados::IoCtx;
+
+namespace librbd {
+namespace image {
+
+template <typename ImageCtxT = ImageCtx>
+class CloneRequest {
+public:
+  static CloneRequest *create(ImageCtx *p_imctx, IoCtx &c_ioctx, const std::string &c_name,
+			      ImageOptions c_options,
```

Comment:
Ok, agree we need a copy constructor. Still, I would like the options to be passed by (const) reference here, to be consistent with CreateRequest. It will still be copied to m_opts in the constructor.